staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -124,6 +125,9 @@ type Config struct {
 	// allows one to setup monitoring and alerting of unexpected rotation
 	// behavior and track trends in rotation frequency.
 	CertificateRotation Histogram
+	// CertifcateRenewFailure will record a metric that keeps track of
+	// certificate renewal failures.
+	CertificateRenewFailure CounterVec
```

Comment:
Maybe change this to:

```
CertificateRenewFailure func(...string)
```
and pass in:
```
func(a) {
  metric.WithLabels(a...).Inc()
}
```
From kubelet so client-go doesn't need to depend on "k8s.io/component-base/metrics".

Reply:
yeah, need to prevent this:
```
tsort: cycle in data
tsort: k8s.io/client-go
tsort: k8s.io/component-base
``